src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Connections.Abstractions.Features;
+using Microsoft.AspNetCore.Connections.Features;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3.QPack;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3
+{
+    internal class Http3Connection : IRequestProcessor
+    {
+        public HttpConnectionContext Context { get; private set; }
+
+        public DynamicTable DynamicTable { get; set; }
+
+        public Http3ControlStream SettingsStream { get; set; }
+        public Http3ControlStream EncoderStream { get; set; }
+        public Http3ControlStream DecoderStream { get; set; }
+
+        private readonly ConcurrentDictionary<int, Http3Stream> _streams = new ConcurrentDictionary<int, Http3Stream>();
+
+        public Http3Connection(HttpConnectionContext context)
+        {
+            Context = context;
+            DynamicTable = new DynamicTable(0);
+            // TODO qpack decoders need to be one per stream?
+
+        }
+
+        public async Task ProcessRequestsAsync<TContext>(IHttpApplication<TContext> application)
+        {
+            var streamFeature = Context.ConnectionFeatures.Get<IQuicStreamListenerFeature>();
+            if (streamFeature == null)
+            {
+                throw new Http3ConnectionException();
+            }
+
+            try
+            {
+                var i = 0;
+                while (true)
+                {
+                    var connectionContext = await streamFeature.AcceptAsync();
+                    if (connectionContext == null)
+                    {
+                        break;
+                    }
+
+                    // Need to keep calling AcceptAsync here.
+                    var httpConnectionContext = new HttpConnectionContext
+                    {
+                        ConnectionId = connectionContext.ConnectionId + i, // TODO this is just adding an int for the stream id, which isn't great.
+                        ConnectionContext = connectionContext,
+                        Protocols = Context.Protocols,
+                        ServiceContext = Context.ServiceContext,
+                        ConnectionFeatures = connectionContext.Features,
+                        MemoryPool = Context.MemoryPool,
+                        Transport = connectionContext.Transport,
+                        TimeoutControl = Context.TimeoutControl,
+                        LocalEndPoint = connectionContext.LocalEndPoint as IPEndPoint,
+                        RemoteEndPoint = connectionContext.RemoteEndPoint as IPEndPoint
```

Comment:
These should always be `IPEndPoint` with QUIC, right? Can you `(IPEndPoint)` instead?

Reply:
When we start testing in memory, it'll probably be MemoryEndPoint or something similar.